mlflow/langchain/__init__.py

```diff
@@ -393,14 +438,46 @@ def _save_model(model, path):
     return model_data_kwargs
 
 
-def _load_model(path, agent_path=None, tools_path=None, agent_primitive_path=None):
-    model = None
-    if agent_path is None and tools_path is None:
-        from langchain.chains.loading import load_chain
+def _load_model(
+    path,
+    model_type,
+    agent_path=None,
+    tools_path=None,
+    agent_primitive_path=None,
+    loader_fn_path=None,
+    persist_dir=None,
+):
+    from langchain.chains.loading import load_chain
 
+    model = None
+    if model_type == "RetrievalQA":
+        with open(loader_fn_path, "rb") as f:
+            loader_fn = cloudpickle.load(f)
+        retriever = loader_fn(persist_dir)
+        model = load_chain(path, retriever=retriever)
```

Comment:
What happens if `loader_fn_path` and `persist_dir` are `None` in these cases? I think we should throw a clear exception

Reply:
`persist_dir` can be `None`, if `loader_fn` does not take any arguments. `loader_fn_path` usually won't be `None` since mlflow generates the path during logging. I'll add the the exception saying something like "model file corruption".